packages/core/src/utils/zod.ts

```diff
@@ -1,20 +1,98 @@
 import { arbitraryObjectGuard } from '@logto/schemas';
-import { conditional } from '@silverhand/essentials';
+import { conditional, ValuesOf } from '@silverhand/essentials';
 import { OpenAPIV3 } from 'openapi-types';
 import {
   ZodArray,
   ZodBoolean,
+  ZodLiteral,
   ZodNativeEnum,
   ZodNullable,
   ZodNumber,
   ZodObject,
   ZodOptional,
   ZodString,
+  ZodStringDef,
+  ZodType,
+  ZodUnion,
   ZodUnknown,
 } from 'zod';
 
 import RequestError from '@/errors/RequestError';
 
+type ZodStringCheck = ValuesOf<ZodStringDef['checks']>;
+
+const zodStringCheckToSwaggerFormat = (zodStringCheck: ZodStringCheck) => {
+  const { kind } = zodStringCheck;
+
+  switch (kind) {
+    case 'email':
+    case 'url':
+    case 'uuid':
+    case 'cuid':
+    case 'regex':
+      return kind;
+
+    case 'min':
+    case 'max':
+      // Do nothing here
+      return;
+
+    default:
+      throw new RequestError('swagger.invalid_zod_type', zodStringCheck);
```

Comment:
A test case can be added to cover this fragment

Reply:
added the test case. PTAL